pkg/operator/workload/helper.go

```diff
@@ -13,10 +14,18 @@ import (
 )
 
 func SkipProcessing(ctx context.Context, resource client.Object, or kube.ObjectResolver, scanOnlyCurrentRevisions bool, log logr.Logger) (bool, error) {
-	controller := metav1.GetControllerOf(resource)
-	switch resource.(type) {
+	switch r := resource.(type) {
 	case *appsv1.ReplicaSet:
+		_, err := or.GetActivePodsMatchingLabels(ctx, resource.GetNamespace(), r.Spec.Selector.MatchLabels)
+		if err != nil {
+			if errors.Is(err, kube.ErrNoRunningPods) {
+				log.V(1).Info("Ignoring ReplicaSet with no active pods", "name", resource.GetName())
+				return true, nil
+			}
+			return true, err
+		}
```

Comment:
This piece of code is getting repeated 3 times, perhaps we can refactor later and extract it to a func.